client/dashboard/sites/site/environment-switcher.tsx

```diff
@@ -148,16 +163,38 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
 	const queryClient = useQueryClient();
 
 	const productionSiteId = getProductionSiteId( site );
-	const stagingSiteId = getStagingSiteId( site );
 
 	const { data: productionSite } = useQuery( {
 		...siteByIdQuery( productionSiteId ?? 0 ),
 		enabled: !! productionSiteId,
 	} );
 
+	const stagingSiteId = getStagingSiteId( productionSite ?? site );
+
+	const { data: isStagingSiteCreating } = useQuery( {
+		...isCreatingStagingSiteQuery( productionSiteId ?? 0 ),
+		enabled: !! productionSiteId,
+	} );
+
+	const { data: atomicTransfer } = useQuery( {
+		...siteLatestAtomicTransferQuery( stagingSiteId ?? 0 ),
+		refetchInterval: ( query ) => {
+			return isAtomicTransferInProgress( query.state.data?.status ?? 'pending' ) ? 2000 : false;
+		},
+		enabled: isStagingSiteCreating && !! stagingSiteId,
+	} );
+	const transferStatus = atomicTransfer?.status;
+
 	const { data: stagingSite } = useQuery( {
 		...siteByIdQuery( stagingSiteId ?? 0 ),
-		enabled: !! stagingSiteId,
+		refetchInterval: ( query ) => {
+			if ( ! query.state.data ) {
+				return 0;
+			}
+
+			return ! isAtomicTransferredSite( query.state.data ) ? 2000 : false;
```

Comment:
Maybe we could considerer swapping the order of `2000` and `false` to avoid negating `isAtomicTransferredSite` (for better readability).

```suggestion
			return isAtomicTransferredSite( query.state.data ) ? false : 2000;
```

Reply:
Changed as part of [acf9ae3](https://github.com/Automattic/wp-calypso/pull/105518/commits/acf9ae37b57c6c02ce473940b91acbb29a5a5d3f)